src/main/java/com/spotify/docker/client/LogReader.java

```diff
@@ -77,4 +91,49 @@ public void close() throws IOException {
     // where close is a no-op.
     copy(stream, nullOutputStream());
   }
+
+  protected synchronized int readWithTimeout(final InputStream in, final byte[] buff,
+      final int off, final int len) {
+    Future<Integer> readCount =  readStreamExecutorService.submit(new Callable<Integer>() {
+      @Override
+      public Integer call() throws Exception {
+        int count = LogReader.this.read(in, buff, off, len);
+        return count;
+      }
+    });
+
+    try {
+      int readCountInt = readCount.get(readTimeoutInMilliseconds, TimeUnit.MILLISECONDS);
+      return readCountInt;
+    } catch (Exception theException) {
+      System.currentTimeMillis();
+    }
+
+    readStreamExecutorService.shutdown();
+    readStreamExecutorService = Executors.newSingleThreadExecutor();
+    return 0;
```

Comment:
why shutdown and re-initialize the executor in case of exception, and then return 0?